scalding-core/src/main/scala/com/twitter/scalding/typed/TypedPipe.scala

```diff
@@ -471,7 +471,18 @@ trait TypedPipe[+T] extends Serializable {
    * Reasonably common shortcut for cases of total associative/commutative reduction
    * returns a ValuePipe with only one element if there is any input, otherwise EmptyValue.
    */
-  def sum[U >: T](implicit plus: Semigroup[U]): ValuePipe[U] = ComputedValue(groupAll.sum[U].values)
+  def sum[U >: T](implicit plus: Semigroup[U]): ValuePipe[U] = {
+    // every 1000 items, compact.
+    lazy implicit val batchedSG = Batched.compactingSemigroup[U](1000)
+    ComputedValue(map { t => ((), Batched[U](t)) }
+      .sumByLocalKeys
+      // remove the Batched before going to the reducers
+      .map { case (_, batched) => batched.sum }
```

Comment:
Speaking for myself, I'm good with this change - I agree that 1000 values should be fine in this case where we're limited to a single key.

Separately (and for a later PR), I'm excited about the idea of integrating `Batched` into `AdaptiveCache`, because it seems like it holds some promise of extending this optimization to more general cases (that is, if there is > 1 key, but still a small number of them relative to the size of the values, it would be nice to get the batching automatically).